frontend/src/app/user.service.spec.ts

```diff
@@ -23,101 +26,179 @@ describe('UserService', () => {
+
+    expect(actualUser).toEqual(expectedUser);
+  });
+
+  it('should update a user', () => {
+    const command: UserCommand = { login: 'jb' } as UserCommand;
+    service.update(2, command).subscribe(() => {});
+
+    const testRequest = http.expectOne({ url: '/api/users/2', method: 'PUT' });
+    expect(testRequest.request.body).toEqual(command);
+    testRequest.flush(null);
+
+    http.verify();
+  });
+
+  it('should create a user', () => {
+    const command: UserCommand = { login: 'jb' } as UserCommand;
+    const expectedUser: UserWithPasswordModel = { id: 2 } as UserWithPasswordModel;
+
+    let actualUser;
+    service.create(command).subscribe(user => actualUser = user);
+
+    const testRequest = http.expectOne({ url: '/api/users', method: 'POST' });
+    expect(testRequest.request.body).toEqual(command);
+    testRequest.flush(expectedUser);
+
+    expect(actualUser).toEqual(expectedUser);
+  });
+
+  it('should list users', () => {
+    const expectedUsers: Array<UserModel> = [{ id: 1 }] as Array<UserModel>;
+
+    let actualUsers;
+    service.list().subscribe(users => actualUsers = users);
+
+    http.expectOne({url: '/api/users', method: 'GET'}).flush(expectedUsers);
+
+    expect(actualUsers).toEqual(expectedUsers);
+  });
+
+  it('should reset a user password', () => {
+    const expectedUser: UserWithPasswordModel = { id: 2 } as UserWithPasswordModel;
+
+    let actualUser;
+    service.resetPassword(2).subscribe(user => actualUser = user);
+
+    const testRequest = http.expectOne({ url: '/api/users/2/password-resets', method: 'POST' });
+    expect(testRequest.request.body).toBeNull();
+    testRequest.flush(expectedUser);
+
+    expect(actualUser).toEqual(expectedUser);
+  });
+
+  it('should change the current user password', () => {
+    service.changePassword('secret').subscribe(() => {});
+
+    const testRequest = http.expectOne({url: '/api/users/me/passwords', method: 'PUT'});
+    expect(testRequest.request.body).toEqual({ newPassword: 'secret' });
+    testRequest.flush(null);
+
+    http.verify();
+  });
+
+  it('should check the current user password', () => {
+    service.userEvents.next({login: 'jb'} as UserModel);
+
+    let ok = false;
+    service.checkPassword('secret').subscribe(() => ok = true);
+
+    const testRequest = http.expectOne({ url: '/api/authentication', method: 'POST' });
+    expect(testRequest.request.body).toEqual({login: 'jb', password: 'secret'});
+    testRequest.flush({id: 2});
+
+    expect(ok).toBe(true);
+  });
+
+  it('should check the current user password with failure', () => {
+    service.userEvents.next({login: 'jb'} as UserModel);
+
+    let ok = false;
+    service.checkPassword('secret').subscribe(null, () => ok = true);
+
+    const testRequest = http.expectOne({ url: '/api/authentication', method: 'POST' });
+    expect(testRequest.request.body).toEqual({login: 'jb', password: 'secret'});
+    testRequest.flush(null, { status: 401, statusText: 'Unauthorized' });
```

Comment:
I saw, but it isn't used to send back a response with an error status code, but rather an error event (simulating an interceptor returning it, or a network failure, or something like that).